### PR TITLE
fix: Strip file extension in attachment parameter for data export

### PR DESCRIPTION
**Pull request by mortenoh**
Jira Issue [DHIS2-11843](https://jira.dhis2.org/browse/DHIS2-11843)

### Summary

Our import/export app is currently sending in `attachment=` parameter with extension included which is causing the backend to add the extension twice, so `attachment=data.xml.zip` ends up `data.xml.zip.xml.zip`, this PR cleans this up and make sure that the correct extension is added (including for files inside of zip).

### CODE REVIEW

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/DataValueSetController.java [hunk 1]
```diff
@@ -334,19 +367,24 @@ private OutputStream compress( HttpServletResponse response, String attachment,
...
+
             response.setHeader( ContextUtils.HEADER_CONTENT_DISPOSITION,
-                "attachment; filename=" + fileName + "." + format + ".gzip" );
+                "attachment; filename=" + fileName + "." + format + ".gz" );
             response.setHeader( ContextUtils.HEADER_CONTENT_TRANSFER_ENCODING, "binary" );
             return new GZIPOutputStream( response.getOutputStream() );
         }
         else if ( Compression.ZIP == compression )
         {
+            fileName = fileName.replace( "." + format + ".zip", "" );
```
Comment (jbee) [inline dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/DataValueSetController.java:380]: Cannot quite follow the idea here. If the requested file already has a e.g. `xyz.json.zip` in the name we strip that part?
Then later we use `fileName` again and append `.json` so we send the file originally named `xyz.json.zip` as `xyz.json`? Something does not match up for me here. But that might be me not getting the intention here.

Comment (mortenoh) [reply dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/DataValueSetController.java:380]: The client shouldn't really be sending in `.xml.zip` at all, but our current import/export app does exactly that. So it seemed cleaner to make sure we don't have any additional suffix (except for wanted filename) and then add it back where it makes sense. We could of course check for extension etc, but this seems like a simpler solution.

A more appropriate use of this endpoint is to send `attachment=fileName&compression=zip&format=json` and then we take care of the rest (including naming it appropriately in the zip file).

**Review by jbee — APPROVED** [review]

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/DataValueSetController.java [hunk 2]
```diff
@@ -116,6 +116,39 @@
...
+        DataValueSetQueryParams params,
+        @RequestParam( required = false ) String attachment,
+        @RequestParam( required = false ) String compression,
+        @RequestParam( required = false ) String format,
+        IdSchemes idSchemes, HttpServletResponse response )
+        throws IOException
+    {
+        setNoStore( response );
+
+        if ( "xml".equals( format ) )
```
Comment (larshelge) [inline dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/DataValueSetController.java:130]: Use some sort of constant for xml and csv instead?

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/DataValueSetController.java (hunk 1 above)
Comment (larshelge) [inline dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/DataValueSetController.java:380]: Could we move this particular operation to a util method with Javadoc and a test? Just to improve test coverage and to make the code more readable/make it easier to understand what is going on.

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/render/RenderFormat.java [hunk 3]
```diff
@@ -33,5 +33,21 @@
...
+
+    public static boolean isXml( String format )
+    {
+        return XML.name().equalsIgnoreCase( format );
+    }
+
+    public static boolean isCsv( String format )
+    {
+        return CSV.name().equalsIgnoreCase( format );
+    }
```
Comment (jbee) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/render/RenderFormat.java:52]: Having an enum isn't it more natural to have a single method that is used like this: `XML.isEqual(format)` ?

**Review by jbee — APPROVED** [review]

**Review by larshelge — APPROVED** [review]

**Review by jbee — APPROVED** [review]

**Pull request merged**